src/cohere/bedrock_client.py

```diff
@@ -33,229 +21,15 @@ def __init__(
-                if "event_type" in streamed_obj:
-                    response_type = stream_response_mapping[endpoint]
-                    parsed = typing.cast(response_type,  # type: ignore
-                                         construct_type(type_=response_type, object_=streamed_obj))  # type: ignore
-                    yield (json.dumps(parsed.dict()) + "\n").encode("utf-8")  # type: ignore
-
-
-def map_response_from_bedrock(
-        chat_model: typing.Optional[str] = None,
-        embed_model: typing.Optional[str] = None,
-        generate_model: typing.Optional[str] = None,
-):
-    def _hook(
-            response: httpx.Response,
-    ) -> None:
-        stream = response.headers["content-type"] == "application/vnd.amazon.eventstream"
-        endpoint = get_endpoint_from_url(response.url.path, chat_model, embed_model, generate_model)
-        output: typing.Iterator[bytes]
-
-        if stream:
-            output = stream_generator(httpx.Response(
-                stream=response.stream,
-                status_code=response.status_code,
-            ), endpoint)
-        else:
-            response_type = response_mapping[endpoint]
-            output = iter([json.dumps(typing.cast(response_type,  # type: ignore
-                                                  construct_type(
-                                                      type_=response_type,
-                                                      object_=json.loads(response.read()))).dict()  # type: ignore
-                                      ).encode(
-                "utf-8")])
-
-        response.stream = Streamer(output)
-        response.is_stream_consumed = False
-        response.is_closed = False
-
-    return _hook
-
-
-def map_request_to_bedrock(
-        service: str,
-        aws_access_key: typing.Optional[str] = None,
-        aws_secret_key: typing.Optional[str] = None,
-        aws_session_token: typing.Optional[str] = None,
-        aws_region: typing.Optional[str] = None,
-        chat_model: typing.Optional[str] = None,
-        embed_model: typing.Optional[str] = None,
-        generate_model: typing.Optional[str] = None,
-) -> EventHook:
-    session = boto3.Session(
-        region_name=aws_region,
-        aws_access_key_id=aws_access_key,
-        aws_secret_access_key=aws_secret_key,
-        aws_session_token=aws_session_token,
-    )
-    credentials = session.get_credentials()
-    signer = SigV4Auth(credentials, service, session.region_name)
-
-    model_lookup = {
-        "embed": embed_model,
-        "chat": chat_model,
-        "generate": generate_model,
-    }
-
-    def _event_hook(request: httpx.Request) -> None:
-        headers = request.headers.copy()
-        del headers["connection"]
-
-        endpoint = request.url.path.split("/")[-1]
-        body = json.loads(request.read())
-
-        url = get_url(
-            platform=service,
+            service="bedrock",
```

Comment:
nit: should we have an enum for the service names?

Reply:
oops i merged before i saw this but yeah I'll follow up